src/dialog/foundation.tsx

```diff
@@ -162,6 +162,9 @@ export const useDialogFoundation = (
         foundation.close();
       }
     }
+    return () => {
+      document.removeEventListener('keydown', handleDocumentKeydown);
+    };
```

Comment:
This corrects issue #781. When the props change, we need to unbind the existing handlers.